packages/ra-ui-materialui/src/list/datagrid/DatagridBody.tsx

```diff
@@ -34,31 +34,33 @@ const DatagridBody: FC<DatagridBodyProps> = React.forwardRef(
             className={clsx('datagrid-body', className, DatagridClasses.tbody)}
             {...rest}
         >
-            {data.map((record, rowIndex) =>
-                cloneElement(
-                    row,
-                    {
-                        className: clsx(DatagridClasses.row, {
-                            [DatagridClasses.rowEven]: rowIndex % 2 === 0,
-                            [DatagridClasses.rowOdd]: rowIndex % 2 !== 0,
-                        }),
-                        expand,
-                        hasBulkActions: hasBulkActions && !!selectedIds,
-                        hover,
-                        id: record.id ?? `row${rowIndex}`,
-                        key: record.id ?? `row${rowIndex}`,
-                        onToggleItem,
-                        record,
```

Comment:
This is a breaking change: a custom DatagridRow will no longer receive the record. You must document it in the upgrade guide.